src/ItemVirtualMachine.php

```diff
@@ -282,98 +274,69 @@ public static function showForAsset(CommonDBTM $asset)
-                echo "<td>" . Dropdown::getYesNo($vm->isDynamic()) . "</td>";
-                echo "<td>";
-                echo Dropdown::getDropdownName(
-                    'glpi_virtualmachinetypes',
-                    $virtualmachine['virtualmachinetypes_id']
-                );
-                echo "</td>";
-                echo "<td>";
-                echo Dropdown::getDropdownName(
-                    'glpi_virtualmachinesystems',
-                    $virtualmachine['virtualmachinesystems_id']
-                );
-                echo "</td>";
-                echo "<td>";
-                echo Dropdown::getDropdownName(
-                    'glpi_virtualmachinestates',
-                    $virtualmachine['virtualmachinestates_id']
-                );
-                echo "</td>";
-                echo "<td>" . $virtualmachine['uuid'] . "</td>";
-                echo "<td>" . $virtualmachine['vcpu'] . "</td>";
-                echo "<td>" . $virtualmachine['ram'] . "</td>";
-                echo "<td>";
-                if ($link_asset = self::findVirtualMachine($virtualmachine)) {
-                     $asset = new $itemtype();
-                    if ($asset->can($link_asset, READ)) {
-                        $url  = "<a href='" . $asset->getFormURLWithID($link_asset) . "'>";
-                        $url .= $asset->fields["name"] . "</a>";
-
-                        $tooltip = "<table><tr><td>" . __('Name') . "</td><td>" . $asset->fields['name'] .
-                            '</td></tr>';
-                        if (isset($asset->fields['serial'])) {
-                            $tooltip .= "<tr><td>" . __('Serial number') . "</td><td>" . $asset->fields['serial'] .
-                                '</td></tr>';
-                        }
-                        if (isset($asset->fields['comment'])) {
-                            $tooltip .= "<tr><td>" . __('Comments') . "</td><td>" . $asset->fields['comment'] .
-                                '</td></tr></table>';
-                        }
-
-                        $url .= "&nbsp; " . Html::showToolTip($tooltip, ['display' => false]);
-                    } else {
-                        $url = $asset->fields['name'];
-                    }
-                    echo $url;
-                }
-                echo "</td>";
-                echo "</tr>";
-                Session::addToNavigateListItems('ItemVirtualMachine', $virtualmachine['id']);
+
+                $type = VirtualMachineType::getById($virtualmachine['virtualmachinetypes_id']);
+                $system = VirtualMachineSystem::getById($virtualmachine['virtualmachinesystems_id']);
+                $state = VirtualMachineState::getById($virtualmachine['virtualmachinestates_id']);
+
+                $entries[] = [
+                    'name'                      => $vm->getLink() ?? NOT_AVAILABLE,
+                    'comment'                   => $virtualmachine['comment'] ?? NOT_AVAILABLE,
```

Comment:
There is no reason to have null/undefined values here.
```suggestion
                    'name'                      => $vm->getLink(),
                    'comment'                   => $virtualmachine['comment'],
```